cmake/BALLConfigBoost.cmake

```diff
@@ -1,55 +1,31 @@
-SET(BALL_BOOST_COMPONENTS system thread iostreams regex date_time serialization chrono)
-if ( WIN32 )
-  LIST(APPEND BALL_BOOST_COMPONENTS zlib bzip2)
-	ADD_DEFINITIONS(-DBOOST_ALL_DYN_LINK)
-	ADD_DEFINITIONS(-DBOOST_THREAD_USE_DLL)
-endif()
-SET(Boost_ADDITIONAL_VERSIONS "1.39" "1.39.0" "1.40" "1.40.0" "1.41" "1.41.0"
-        "1.42" "1.42.0" "1.43" "1.43.0" "1.44.0" "1.45.0" "1.46.0" "1.46.1" "1.47" "1.47.0" "1.47.1" "1.59")
-SET(Boost_DETAILED_FAILURE_MSG ON)
-FIND_PACKAGE(Boost COMPONENTS ${BALL_BOOST_COMPONENTS})
+# Find and configure Boost library
 
-IF (NOT Boost_VERSION) ## we cannot test for Boost_FOUND since this requires all components
-	MESSAGE(SEND_ERROR "Could not find a suitable boost installation! This is a required dependency for BALL! Try setting BOOST_ROOT in ccmake!")
-ELSE()
-	SET(BOOST_LIBRARIES "")
-	FOREACH(COMPONENT ${BALL_BOOST_COMPONENTS})
-		STRING(TOUPPER ${COMPONENT} COMPONENT)
-		
-		IF (Boost_${COMPONENT}_FOUND)
-			SET(BALL_HAS_BOOST_${COMPONENT} TRUE)
-			LIST(APPEND BOOST_LIBRARIES ${Boost_${COMPONENT}_LIBRARY})
-		ENDIF()
-	ENDFOREACH()
-	
-	## For asio, we have some more work to do...
-	MESSAGE(STATUS ${Boost_INCLUDE_DIRS})
-	FIND_PATH(BOOST_ASIO_DIR 
-		NAMES "boost/asio.hpp"
-		PATHS ${Boost_INCLUDE_DIRS})
+# Mandatory boost components
+SET(BALL_BOOST_COMPONENTS
+	bzip2
+	chrono
+	date_time
+	iostreams
+	regex
+	serialization
+	system
+	thread
+	zlib
+)
```

Comment:
To be honest, I haven't checked that. I've just took the ones we already had included.
If we don't need chrono and date_time we can get rid of them.

With the exception of bzip2/zlib: I included those because once we stumbled over the problem that using boost::iostreams compressor filters (I think it was the MolFileFactory) led to a run-time problem because of missing bzip2 libraries. Thus, I included them in the contrib and also here.